### DISCUSSION TITLE
Sync concept of learned ham and spam over multiple servers

### DISCUSSION THREAD

**Discussion by lbayerlein** [rspamd/rspamd#3653] [Q&A]
Hi all,

I am administrating 3 email postfix based email servers. This week, I installed rspamd and it such a great tool. Now I would like to connect learnings of ham and spam from all 3 servers and serve it to rspamd to fight against spam.

Unfortunately, I do not know hot to do it. I created a new server called "mxgate", that should collect all ham and spam informations and share it with all email servers. In my opinion, clients should learn on their base and share it with central (mxgate). New mails should be verified from informations of the mxgate.
On the client side, a redis slave synchronises its database to the master (mxgate). If I take a look into the file structure, the databases are growing - this seems fine. If I take a look into the webui, no learnings are displayed.

What I am doing wrong?

Thank you very much

**Comment by moisseev** [comment]
> Unfortunately, I do not know hot to do it. I created a new server called "mxgate", that should collect all ham and spam informations and share it with all email servers. In my opinion, clients should learn on their base and share it with central (mxgate). New mails should be verified from informations of the mxgate.

It should work, but it is not necessary to set up another server. You can just replicate Redis database from one of the 3 existing servers to other 2 and write learning statistics from slaves directly to the master Redis.

The details explained in this article: https://rspamd.com/doc/tutorials/redis_replication.html

> If I take a look into the webui, no learnings are displayed.
What I am doing wrong?

It's hard to guess. Try to compare `rspamc stat` output on the master and a slave.

**Reply by lbayerlein** [reply]
Oh ok I think, I was confused. Thank you very much for your answer. Now it works very well and the learned statistics are shown in `rspamc stat` and the webui. What I have done wrong: I forgot to add the classifier-bayes on the master and had a missunderstanding, whats slave and whats master in case of redis synchronization.

Sorry and thanks again! For me, it is solved :)
